agreementbot/api.go

```diff
@@ -112,6 +112,143 @@ func (a *API) GetName() string {
 	return a.name
 }
 
+type ServedOrgs struct {
+	ServedPatterns map[string]exchange.ServedPattern        `json:"servedPatterns"`
+	ServedPolicies map[string]exchange.ServedBusinessPolicy `json:"servedPolicies"`
+}
+
+func (a *API) ListServedOrgs(w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case "GET":
+		// retrieve info to write
+		info := ServedOrgs{ServedPatterns: patternManager.ServedPatterns, ServedPolicies: businessPolManager.ServedPolicies}
```

Comment:
You cant access the pattern manager or the businessPolManager fields without the appropriate concurrency lock. You might have to add a getter method to each manager that gets the lock before returning the served org map, and it will have to return a copy of the map. You need to check every other field access from this new code. I haven't checked them all.

Reply:
Done. Added some basic getters that return a copy of the fields and replaced all the field accesses with calls to the getter.